Cargo.toml

```diff
@@ -15,30 +16,41 @@ license    = "MIT"
 repository = "https://github.com/cosmos/solidity-ibc-eureka"
 keywords   = ["cosmos", "ibc", "sp1", "tendermint", "ethereum", "bridge", "solidity", "eureka"]
 
+[workspace.metadata.scripts]
+build-08-wasm-eth = """docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/optimizer:0.16.1 ./programs/08-wasm-eth"""
+
 [workspace.dependencies]
 ibc-eureka-solidity-types          = { path = "packages/solidity", default-features = false }
 ibc-eureka-relayer-lib             = { path = "packages/relayer-lib", default-features = false }
-sp1-ics07-tendermint-prover        = { path = "packages/prover", default-features = false }
-sp1-ics07-tendermint-utils         = { path = "packages/utils", default-features = false }
+sp1-ics07-tendermint-prover        = { path = "packages/sp1-ics07-tendermint-prover", default-features = false }
+sp1-ics07-tendermint-utils         = { path = "packages/sp1-ics07-tendermint-utils", default-features = false }
 sp1-ics07-tendermint-update-client = { path = "programs/sp1-programs/update-client", default-features = false }
 sp1-ics07-tendermint-membership    = { path = "programs/sp1-programs/membership", default-features = false }
+ethereum-trie-db                   = { path = "packages/ethereum-trie-db", default-features = false }
+ethereum-utils                     = { path = "packages/ethereum-utils", default-features = false }
+ethereum-light-client              = { path = "packages/ethereum-light-client", default-features = false }
 
-serde           = { version = "1.0", default-features = false }
-serde_json      = { version = "1.0", default-features = false }
+serde           = { version = "1.0", default-features = false, features = ["derive"] }
+serde_json      = { version = "1.0", default-features = false, features = ["alloc"] }
```

Comment:
Let's not introduce any features at the top level `Cargo.toml`. By following this approach, I was able to remove some deps and even precompiles. Can't these features be imported in lower level crates?

Reply:
Ah, yeah that is nice. I'll do that 👍